recipe/meta.yaml

```diff
@@ -57,8 +58,8 @@ requirements:
     - {{ compiler('cuda') }}              # [cuda_compiler_version != "None"]
     - {{ compiler('c') }}                 # [cuda_compiler_version != "None"]
     - {{ stdlib('c') }}                   # [cuda_compiler_version != "None"]
-    - python                              # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - python                              # [cuda_compiler_version != "None" and build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [cuda_compiler_version != "None" and build_platform != target_platform]
```

Comment:
This change is wrong. even when cuda compiler version is none we need the cross-python tools for cross-compiled builds of the python package in this recipe.

Reply:
See the osx-arm64 build failures with this change for example.

Reply:
Yeah was just trying stuff. Agree this isn't working

Should we consider moving this [to `noarch` using virtual OS packages]( https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-packages-with-os-specific-dependencies )?

Reply:
We can but i'd like to understand why this is failing when it did not this morning.

Reply:
Also, do you know why we need all of the cuda variants?

Reply:
Agreed. It is odd

The CUDA variants are for testing ( https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/93 ). There have been a few rounds of cleanup of these ( for example: https://github.com/conda-forge/conda-forge-ci-setup-feedstock/issues/191 ). At this point only CUDA pre-12 uses this, which is only CUDA 11.8. It will go away whenever we drop CUDA 11.8